docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+# Multi-tenanted solution architecture
+In this document, we’re going to discuss concepts in multi-tenanted solution architecture.
+
+We will talk about
+- what multi-tenanted means
+- when to choose a multi-tenanted architecture
+- tenancy as a hierarchy
+- the management plane and the solution plane
+- how tenancy impacts application code and resource choices
+- how Corvus.Tenancy and Marain.Tenancy help with building multi-tenanted services.
+
+## What does multi-tenanted mean?
+Let’s start by thinking about the distinction between *multi-tenant* and *multi-user*.
+
+That's the first challenge. There is no hard-and-fast definition of what *multi-tenant* even means. So we need to define the problem a bit more clearly.
+
+The chief problem with giving multiple users access to a computer is that we need a fair way of ensuring that they don’t tread on each other’s toes.
+
+Initially, people used to book time on the computer. They would get exclusive access to a giant room of glowing valves, for a specific period of time. Then get physically booted out.
+
+By the 1970s, the slightly confusingly named [time sharing](https://en.wikipedia.org/wiki/Time-sharing) model predominated. Multiple users could access a single computer at the same time, sharing the resources.
+
+The security and fair-shares implications of this ultimately led to the development of isolated process models, virtual machines, containers, and the cloud/edge computing model we know today. 
+
+As part of this shift, the line between "computer user" and "solution user" has also blurred.
+
+In a modern web application, for example, there is frequently a single logical solution serving hundreds, thousands, or millions of concurrent users.
+
+That solution is likely implemented using many compute and storage resources, distributed across various physical and virtual machines, but it is conceptually *one solution, lots of users*. To developer and user alike, it looks like a single set of resources, serving many users. Those users may work together in logical groups (like teams, or organizations) but they are fundamentally sharing a single solution.
+
+Multi-tenancy, on the other hand, can be thought of as *lots of solutions, lots of users*. To the *owner* of the multi-tenanted solution provider, it looks conceptually like *many isolated child solutions* (or *tenants*) consuming the resources it provides, each of which independently serves many users. To any individual tenant, their "tenancy" looks like a single set of resources that they use to serve their users, and no-one else’s.
+
+![multi-user v. multi tenant](multi-user v multi-tenant.png)
+
+This *conceptual difference* is often confused with the *implementation difference*. It is perfectly possible for the provider of a multi-tenanted service to implement their service using a single set of resources shared between all their tenants (if other constraints permit).
+
+## When should a solution be multi-tenanted, not multi-user?
```

Comment:
A confusing thing about putting it this way is that multi-tenanted systems generally are multi-user. So the reader has to infer an implied "merely" after "not".

Microsoft's recently released (as in, this came out last Friday) SaaS guidance also talks about how some systems have one user is effectively being one tenant. (Some consumer-oriented services work this way.) It goes on to discuss how it's quite common for this to eventually evolve to the point where there might be multiple users per tenant (e.g., consumer-oriented services may want to add a "family" concept where multiple logins have access to the same tenant) but they didn't necessarily start that way.

But even ignoring that "one user is one tenant, for now" case, I think it might be helpful to tweak the language so it's clear that multi-tenant typically includes "multi-user" functionality, by avoiding putting the word "not" right in front of "multi-user".

Reply:
Agreed. That is certainly the intent, and doesn't make it clear.